packages/plugin-apollo-client/src/engine/index.ts

```diff
@@ -7,23 +7,22 @@ import {
 	type ApolloClientOptions,
 	type NormalizedCacheObject,
 	type OperationVariables,
-	type QueryOptions,
 	type ServerError,
 	type ServerParseError,
 } from '@apollo/client';
 import { Logger } from '@snapwp/core';
 import { getGraphqlUrl } from '@snapwp/core/config';
 import { QueryProvider as ApolloQueryProvider } from './query-provider';
 import type { TypedDocumentNode } from '@graphql-typed-document-node/core';
-import type { QueryEngine } from '@snapwp/types';
+import type { QueryEngine, QueryOptions } from '@snapwp/types';
```

Comment:
This (versus the `@apollo/client.QueryOptions`) is what primarily fixed the ts issues after fixing the interface to no longer be shorthand. Needs confirmation I didnt introduce a regression

Reply:
After this change, `fetchQuery` takes options defined by type `@snapwp/types.QueryOptions` regardless of what plugin is installed and configured in the userland. 

@SH4LIN PTAL at this PR and advice on a real fix.

Reply:
@ayushnirwal I've reverted the type changes in https://github.com/rtCamp/snapwp/pull/176/commits/e83a446059317d84a9648a76d01d6b59dff4ba93

@SH4LIN your task is now to remove those `eslint-disable`s I added to the interface, and then update this file to be compliant with the failing eslint rule: https://typescript-eslint.io/rules/method-signature-style/

Reply:
We'll follow this up in a new pr if/as needed